sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6234,6 +6260,44 @@ pi_result piextGetDeviceFunctionPointer(pi_device Device, pi_program Program,
     ModIt++;
   }
 
+  // zeModuleGetFunctionPointer currently fails for all
+  // kernels regardless of if the kernel exist or not
+  // with ZE_RESULT_ERROR_INVALID_ARGUMENT
+  // TODO: remove when this is no longer the case
+  // If zeModuleGetFunctionPointer returns invalid argument,
+  // fallback to searching through kernel list and return
+  // PI_FUNCTION_ADDRESS_IS_NOT_AVAILABLE if the function exists
+  // or PI_INVALID_KERNEL_NAME if the function does not exist.
+  // FunctionPointerRet should always be 0
+  if (ZeResult == ZE_RESULT_ERROR_INVALID_ARGUMENT) {
+    size_t Size;
+    *FunctionPointerRet = 0;
+    pi_result ret_err = piProgramGetInfo(Program, PI_PROGRAM_INFO_KERNEL_NAMES,
+                                         0, nullptr, &Size);
+    if (ret_err != PI_SUCCESS)
+      return ret_err;
+
+    std::string ClResult(Size, ' ');
+    ret_err = piProgramGetInfo(Program, PI_PROGRAM_INFO_KERNEL_NAMES,
+                               ClResult.size(), &ClResult[0], nullptr);
+    if (ret_err != PI_SUCCESS)
+      return ret_err;
+
+    // Get rid of the null terminator and search for kernel_name
+    // If function can be found return error code to indicate it
+    // exists
+    ClResult.pop_back();
```

Comment:
why do you need to remove the null-terminator? looks weird

Reply:
This comes from how it was implemented in program_impl.cpp I ran a quick test removing it. Some tests failed as the kernel cannot be found. I assume an unnecessary character is appended in the kernel list and needs to be removed. 
So I believe this is necessary for the name search to behave correctly.